pandas/core/reshape/pivot.py

```diff
@@ -368,18 +368,32 @@ def _convert_by(by):
 @Appender(_shared_docs['pivot'], indents=1)
 def pivot(data, index=None, columns=None, values=None):
     if values is None:
-        cols = [columns] if index is None else [index, columns]
+        # Make acceptable for multiple column indexes.
+        cols = []
+        if is_list_like(index):
+            cols.extend(index)
```

Comment:
the if/elif/else was fine here; don't create then extend, just create in each block